lib/Graphs/ConsG.cpp

```diff
@@ -666,9 +666,9 @@ struct DOTGraphTraits<ConstraintGraph*> : public DOTGraphTraits<PAG*>
             else if(SVFUtil::isa<FIObjPN>(node))
                 return "shape=septagon";
             else if (SVFUtil::isa<DummyObjPN>(node))
-                return "shape=Mcircle";
+                return "shape=house";
             else
-                return "shape=doublecircle";
+                return "shape=invhouse";
```

Comment:
How about using `pentagon`, `octagon`, and `septagon` for three types of objects?

Reply:
Maybe `doubleoctagon`, `pentagon` and `octagon`?

Reply:
Hi Richard,

Thanks for your summary. Yes, it would be good to distinguish them and make them more readable.
Add a new `getNodeAttributes() const` would be better. Maybe the first option from you is good enough.

I am fine with these shapes 
```
circle (other ValPN) -> box
septagon (FIObjPN) -> box3d
Mcircle (DummyObjPN) -> tab
```

For `GetObjPN` (other ObjPN), can we use `component` which is also good for semantic purposes, since GetObjPN is a field `object` of its base `FIObjPN`.

For node colors, it could make different types more distinguishable, but we also have colors on edges. It might be too fancy for the graph unless you can show us a good example.